test/mysql.test.js

```diff
@@ -235,6 +249,36 @@ describe('mysql', function() {
         });
       });
     });
+
+    it('isNewInstance should be undefined for after save hook', function(done) {
+      var student = {name: 'Joe', age: 20};
+      var newStudent = {};
+      var isNewInstanceBefore = false;
+      var isNewInstanceAfter = false;
+      Student.create(student, function(err, createdStudent) {
+        if (err) return done(err);
+        newStudent.id = createdStudent.id;
+        newStudent.name = 'Hannah';
+        newStudent.age = 25;
+        Student.observe('before save', function(ctx, next) {
+          isNewInstanceBefore = ctx.isNewInstance;
+          next();
+        });
+        Student.observe('after save', function(ctx, next) {
+          isNewInstanceAfter = ctx.isNewInstance;
+          next();
+        });
+        Student.replaceOrCreate(newStudent, function(err, s) {
+          if (err) return done(err);
+          s.id.should.equal(newStudent.id);
+          s.name.should.equal(newStudent.name);
+          s.age.should.equal(newStudent.age);
```

Comment:
I think these three asserts (L273-L275) can be removed - the test is checking how `ctx.isNewInstance` is reported. The fact that `replaceOrCreate` modifies the data correctly is already verified by other tests.